docs/design/proposals/creds-from-kms.md

```diff
@@ -0,0 +1,130 @@
+# Fetching Ceph user credentials from KMS
+
+## Proposal
+
+Ceph-CSI supports several KMS implementations for storing and
+fetching DEKs (Data Encryption Keys) used for RBD volume encryption.
+The focus of this proposal is to leverage existing KMS implementation
+to support obtaining Ceph user keyring used while
+creating, deleting, mapping and resizing RBD volumes.
+
+### Benefits
+
+- For scenarios where Ceph is deployed externally, either as
+  a standalone or through Rook, one needs to obtain the Ceph keyring(s)
+  and manually create the K8s secrets needed by Ceph-CSI. With this feature,
```

Comment:
This is true only if CephCSI is depoyed externally, when Rook deploys the ceph cluster it creates the required secrets for cephcsi.

Reply:
I agree that this feature would directly fit into external setup scenario. However, I would see this as a valid case even when Rook deploys Ceph cluster. I see that Rook creates the following secrets by default, and the user caps are quite broad:
```
rook-csi-cephfs-node
rook-csi-cephfs-provisioner
rook-csi-rbd-node
rook-csi-rbd-provisioner
``` 

However, if we were to support more advanced use case such as soft multi-tenancy. It might make sense to have a ceph users per tenant and provide permissions to only access specific pool(s). 

Also, if Rook plans to add support for https://github.com/rook/rook/issues/6374. It would be a valid case even in normal scenario.

Reply:
>However, if we were to support more advanced use case such as soft multi-tenancy. It might make sense to have a ceph users per tenant and provide permissions to only access specific pool(s). 

Admin can still create new users with different caps using cephclient CRD and use that with custom storageclasses.

Reply:
>This is true only if CephCSI is depoyed externally, when Rook deploys the ceph cluster it creates the required secrets for cephcsi. 

Sorry, I might have misunderstood your comment. 

In this point, I was only referring CephCSI being deployed externally:
>For scenarios where Ceph is deployed externally, either as
  a standalone or through Rook,

`For scenarios where Ceph is deployed externally,` by this, I mean Ceph cluster is not in the same K8s cluster as Ceph-CSI. `either as
  a standalone or through Rook` by this I mean the external Ceph cluster may have been deployed using Rook or some other tool.  

If you feel the sentence isn't clear enough, I can change that a bit. Please let me know.

> Admin can still create new users with different caps using cephclient CRD and use that with custom storageclasses.

Thanks for letting me know. You are right, for Rook based Ceph and CephCSI deployments this approach might not be applicable, unless Rook supports storing keyrings directly in a KMS.